crates/sui-graphql-rpc/src/server/builder.rs

```diff
@@ -96,6 +98,36 @@ impl Server {
             })
         };
 
+        let notify = Arc::new(Notify::new());
+
+        // A handle that checks every 12 hours for an epoch change. If an epoch change is detected,
+        // emits a signal and updates the latest epoch value.
+        let epoch_boundary_task_notifier = {
+            let metrics = self.state.metrics.clone();
+            let cancellation_token = self.state.cancellation_token.clone();
+            let controller = notify.clone();
+            info!("Starting epoch boundary task");
+            spawn_monitored_task!(async move {
+                check_epoch_boundary(
+                    &self.db_reader,
+                    metrics,
+                    tokio::time::Duration::from_secs(12 * 60 * 60),
```

Comment:
I think this is too granular. We probably need to do a check every second, which might be too much, but ideal. How about 30s? That would yield to ~2880 queries per day, which is probably fine.